src/Server.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/**
+ * @file
+ * Contains \Netzstrategen\Ssofact\Server.
+ */
+
+namespace Netzstrategen\Ssofact;
+
+class Server {
+
+  /**
+   * Checks if the email is already registered.
+   *
+   * @param string email
+   *
+   * @return array
+   */
+  public static function checkIsEmailRegistered($email) {
+    $api_url = 'https://stage-login.stimme.de/REST/services/authenticate/user/IsEmailRegistered';
+    $response = wp_remote_post($api_url, [
+      'method' => 'POST',
+      'body' => [
+        'email' => $email
+      ],
+    ]);
+    if ($response instanceof \WP_Error) {
+      trigger_error($response->get_error_message(), E_USER_ERROR);
+      return;
+    }
+    return json_decode($response['body'], JSON_OBJECT_AS_ARRAY);
+  }
+
+  /**
+   * Checks if the subscription ID number matches an existing user.
+   *
+   * @param int subscriptionId
+   * @param string firstName
+   * @param string lastName
+   * @param int zipCode
+   *
+   * @return array
+   */
+  public static function checkSubscriptionNumber($subscriptionId, $firstName, $lastName, $zipCode) {
+    $api_url = 'https://stage-login.stimme.de/REST/services/authenticate/user/checkAboNo';
+    $response = wp_remote_post($api_url, [
+      'method' => 'POST',
+      'body' => [
+        "abono" => $subscriptionId,
+        "firstname" => $firstName,
+        "lastname" => $lastName,
+        "zipcode" => $zipCode,
+      ],
+    ]);
+    if ($response instanceof \WP_Error) {
+      trigger_error($response->get_error_message(), E_USER_ERROR);
+      return;
```

Comment:
(also elsewhere) You also need to return something in case of an unexpected communication error, as you need to prevent the form submission from proceeding.